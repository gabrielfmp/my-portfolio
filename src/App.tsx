import { AboutMe } from './components/about-me';
import {
  Content,
  HorizontalBlock,
  Page,
  SkillsEvaluationWrapper,
  SkillsWrapper,
} from './styles';
import { Experience } from './components/experience';
import { Global } from './Global';
import { Header } from './components/header';
import { SkillEvaluation } from './components/skill-evaluation';
import { Skills } from './components/skills';
import { Title } from './ui';
import React from 'react';

const App = () => {
  return (
    <>
      <Global />

      <Page>
        <Header />

        <Content>
          <HorizontalBlock>
            <Experience />

            <AboutMe />
          </HorizontalBlock>

          <HorizontalBlock>
            <SkillsWrapper>
              <Title>Skills</Title>
              <SkillsEvaluationWrapper>
                <SkillEvaluation skill="Javascript" evaluation={5} outOf={5} />
                <SkillEvaluation skill="HTML" evaluation={3.5} outOf={5} />
                <SkillEvaluation skill="CSS" evaluation={4} outOf={5} />
                <SkillEvaluation skill="React" evaluation={4.5} outOf={5} />
                <SkillEvaluation skill="Redux" evaluation={5} outOf={5} />
              </SkillsEvaluationWrapper>
            </SkillsWrapper>

            <Skills />
          </HorizontalBlock>
        </Content>
      </Page>
    </>
  );
};

export default App;
