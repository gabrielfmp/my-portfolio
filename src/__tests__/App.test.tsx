import { render } from '@testing-library/react';
import App from '../App';
import React from 'react';

test('snapshot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
