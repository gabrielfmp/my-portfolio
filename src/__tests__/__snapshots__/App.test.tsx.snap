// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-59umzr"
  >
    <header
      class="css-1wskg7y"
    >
      <p
        class="css-1ooz59o"
      >
        Hi! I’m a frontend engineer driven by inovation and meaningful connections. I focus mainly on improving every user’s experience with simple yet effective code,
        <s>
          even though most of the time I’m solving bugs I created myself.
        </s>
      </p>
      <h1
        class="css-13fdacc"
      >
        Gabriel
        <br />
        Pires
      </h1>
    </header>
    <div
      class="css-760yb3"
    >
      <div
        class="css-hhdf81"
      >
        <div
          class="css-yp9swi"
        >
          <div
            class="css-ov1ktg"
          >
            <div
              class="css-v5pni"
            />
            <h2
              class="css-111mxsp"
            >
              Education
            </h2>
          </div>
          <div
            class="css-sty4r9"
          >
            <p
              class="css-11bj48"
            >
              Computer Science and Engineering
            </p>
            <div
              class="css-14pnp1n"
            >
              <p
                class="css-zkfum4"
              >
                Instituto Superior Técnico
              </p>
              <div
                class="css-15ag3i8"
              />
              <p
                class="css-zkfum4"
              >
                2015 - 2018
              </p>
            </div>
            <ul
              class="css-1rdmnzn"
            >
              <li
                class="css-1g7zcol"
              >
                Python, C++, Java, C, Assembly, Prolog and more
              </li>
              <li
                class="css-1g7zcol"
              >
                Databases in PostgreSQL
              </li>
              <li
                class="css-1g7zcol"
              >
                Design patterns
              </li>
              <li
                class="css-1g7zcol"
              >
                Algorithms and data structures
              </li>
            </ul>
          </div>
          <div
            class="css-ov1ktg"
          >
            <div
              class="css-v5pni"
            />
            <h2
              class="css-111mxsp"
            >
              Experience
            </h2>
          </div>
          <div
            class="css-15ycokd"
          >
            <p
              class="css-11bj48"
            >
              Frontend Engineer
            </p>
            <div
              class="css-14pnp1n"
            >
              <p
                class="css-zkfum4"
              >
                Farfetch
              </p>
              <div
                class="css-15ag3i8"
              />
              <p
                class="css-zkfum4"
              >
                2018 - Present
              </p>
            </div>
            <ul
              class="css-1rdmnzn"
            >
              <li
                class="css-1g7zcol"
              >
                Learning from experienced developers
              </li>
              <li
                class="css-1g7zcol"
              >
                Creating a platform and enhancing developer experience
              </li>
              <li
                class="css-1g7zcol"
              >
                Working remotely
              </li>
              <li
                class="css-1g7zcol"
              >
                Providing support to Harrods, Off-White and other e-commerce projects
              </li>
              <li
                class="css-1g7zcol"
              >
                Hosting events
              </li>
            </ul>
          </div>
          <div
            class="css-1s6gzqy"
          >
            <div
              class="css-1iyoj2o"
            >
              <div
                class="css-1g6u7np"
              />
              <div
                class="css-1g6u7np"
              />
              <div
                class="css-1g6u7np"
              />
              <div
                class="css-1g6u7np"
              />
              <div
                class="css-1g6u7np"
              />
            </div>
            <p
              class="css-1jtsvsh"
            >
              Next? Your company maybe
            </p>
          </div>
        </div>
        <div
          class="css-1ln1eos"
        >
          <h2
            class="css-2u4su"
          >
            About me
          </h2>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="email"
                class="css-12k99am"
                src="email.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              gabrielfmpires@gmail.com
            </div>
          </div>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="phone number"
                class="css-12k99am"
                src="phone.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              +351 934429387
            </div>
          </div>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="location"
                class="css-12k99am"
                src="location.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              Lisbon, Portugal
            </div>
          </div>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="birthday"
                class="css-12k99am"
                src="cake.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              17 May 1997
            </div>
          </div>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="github"
                class="css-12k99am"
                src="github.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              gabrielfmp
            </div>
          </div>
          <div
            class="css-gqh2np"
          >
            <div
              class="css-47j6j9"
            >
              <img
                alt="linkedin"
                class="css-12k99am"
                src="linkedin.svg"
              />
            </div>
            <div
              class="css-un0t8r"
            >
              gabrielfmpires
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-hhdf81"
      >
        <div
          class="css-yp9swi"
        >
          <h2
            class="css-111mxsp"
          >
            Skills
          </h2>
          <div
            class="css-1b2x0kn"
          >
            <div
              class="css-1tv79r0"
            >
              <p
                class="css-1x26r7c"
              >
                Javascript
              </p>
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
            </div>
            <div
              class="css-1tv79r0"
            >
              <p
                class="css-1x26r7c"
              >
                HTML
              </p>
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-5uorsv"
                color="#BCBCBC"
                data-testid="half-dot"
              >
                <div
                  class="css-1qe1cwe"
                />
              </div>
              <div
                class="css-5uorsv"
                color="#BCBCBC"
                data-testid="empty-dot"
              />
            </div>
            <div
              class="css-1tv79r0"
            >
              <p
                class="css-1x26r7c"
              >
                CSS
              </p>
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-5uorsv"
                color="#BCBCBC"
                data-testid="empty-dot"
              />
            </div>
            <div
              class="css-1tv79r0"
            >
              <p
                class="css-1x26r7c"
              >
                React
              </p>
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-5uorsv"
                color="#BCBCBC"
                data-testid="half-dot"
              >
                <div
                  class="css-1qe1cwe"
                />
              </div>
            </div>
            <div
              class="css-1tv79r0"
            >
              <p
                class="css-1x26r7c"
              >
                Redux
              </p>
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
              <div
                class="css-1dpzy13"
                color="#000"
                data-testid="filled-dot"
              />
            </div>
          </div>
        </div>
        <div
          class="css-1vymck9"
        >
          <div
            class="css-o7xbd4"
          >
            <div
              class="css-1sulisw"
            >
              <p
                class="css-1yo0kyw"
              >
                Critical Thinking
              </p>
              <p
                class="css-1yo0kyw"
              >
                Will to learn
              </p>
            </div>
            <div
              class="css-1sulisw"
            >
              <p
                class="css-1yo0kyw"
              >
                English
              </p>
              <p
                class="css-1yo0kyw"
              >
                Portuguese (native)
              </p>
              <p
                class="css-1yo0kyw"
              >
                Spanish
              </p>
            </div>
            <div
              class="css-1sulisw"
            >
              <p
                class="css-1yo0kyw"
              >
                Communication
              </p>
              <p
                class="css-1yo0kyw"
              >
                Self-improvement
              </p>
            </div>
            <div
              class="css-1sulisw"
            >
              <p
                class="css-1yo0kyw"
              >
                Creativity
              </p>
              <p
                class="css-1yo0kyw"
              >
                Organization
              </p>
              <p
                class="css-1yo0kyw"
              >
                Teamwork
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
